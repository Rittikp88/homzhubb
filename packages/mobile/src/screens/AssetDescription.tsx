import React, { useState } from 'react';
import { FlatList, StyleSheet, TouchableOpacity, View } from 'react-native';
// @ts-ignore
import ParallaxScrollView from 'react-native-parallax-scroll-view';
import { withTranslation, WithTranslation } from 'react-i18next';
import { bindActionCreators, Dispatch } from 'redux';
import { connect } from 'react-redux';
import Collapsible from 'react-native-collapsible';
import MapView, { Marker, PROVIDER_GOOGLE } from 'react-native-maps';
import { PropertyUtils } from '@homzhub/common/src/utils/PropertyUtils';
import { IState } from '@homzhub/common/src/modules/interfaces';
import { AssetActions } from '@homzhub/common/src/modules/asset/actions';
import { AssetSelectors } from '@homzhub/common/src/modules/asset/selectors';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import {
  CustomMarker,
  Divider,
  Label,
  PricePerUnit,
  PropertyAddress,
  Text,
  WithShadowView,
} from '@homzhub/common/src/components';
import { StatusBarComponent } from '@homzhub/mobile/src/components/atoms/StatusBar';
import { AssetRatings } from '@homzhub/mobile/src/components/molecules/AssetRatings';
import ShieldGroup from '@homzhub/common/src/components/molecules/ShieldGroup';
import { AssetDetailsImageCarousel } from '@homzhub/mobile/src/components/molecules/AssetDetailsImageCarousel';
import { FullScreenAssetDetailsCarousel } from '@homzhub/mobile/src/components/molecules/FullScreenAssetDetailsCarousel';
import PropertyAmenities from '@homzhub/mobile/src/components/molecules/PropertyAmenities';
import { Asset } from '@homzhub/common/src/domain/models/Asset';
import { AssetHighlight } from '@homzhub/common/src/domain/models/AssetHighlight';
import { AssetReview } from '@homzhub/common/src/domain/models/AssetReview';
import { IAmenitiesIcons } from '@homzhub/common/src/domain/models/Search';

interface IStateProps {
  reviews: AssetReview[];
  assetDetails: Asset | null;
}

interface IDispatchProps {
  getAssetReviews: (id: number) => void;
}

interface IOwnState {
  isFullScreen: boolean;
  activeSlide: number;
  descriptionShowMore: boolean;
  isScroll: boolean;
}

const { width, height } = theme.viewport;
const realWidth = height > width ? width : height;
const relativeWidth = (num: number): number => (realWidth * num) / 100;

const PARALLAX_HEADER_HEIGHT = 250;
const STICKY_HEADER_HEIGHT = 60;

type Props = IStateProps & IDispatchProps & WithTranslation;
// TODO: Get from redux once set up
const IMAGES = [
  {
    id: 1,
    name: 'image1.png',
    link: 'https://homepages.cae.wisc.edu/~ece533/images/airplane.png',
    attachment_type: 'IMAGE',
    mime_type: 'Image/Jpeg',
    is_cover_image: true,
  },
  {
    id: 2,
    name: 'image2.png',
    link: 'https://www.youtube.com/watch?v=L7OLY4HCctQ',
    attachment_type: 'VIDEO',
    mime_type: 'Image/Jpeg',
    is_cover_image: true,
  },
  {
    id: 3,
    name: 'image3.png',
    link: 'https://homepages.cae.wisc.edu/~ece533/images/baboon.png',
    attachment_type: 'IMAGE',
    mime_type: 'Image/Jpeg',
    is_cover_image: true,
  },
  {
    id: 4,
    name: 'image4.png',
    link: 'https://homepages.cae.wisc.edu/~ece533/images/boat.png',
    attachment_type: 'IMAGE',
    mime_type: 'Image/Jpeg',
    is_cover_image: true,
  },
  {
    id: 5,
    name: 'image5.png',
    link: 'https://homepages.cae.wisc.edu/~ece533/images/monarch.png',
    attachment_type: 'IMAGE',
    mime_type: 'Image/Jpeg',
    is_cover_image: true,
  },
];

interface ICollapsibleSectionProps {
  title: string;
  children: React.ReactNode;
  initialCollapsedValue?: boolean;
}
const CollapsibleSection = (props: ICollapsibleSectionProps): React.ReactElement => {
  const { title, children, initialCollapsedValue = false } = props;
  const [isCollapsed, setIsCollapsed] = useState(initialCollapsedValue);

  const onPress = (): void => {
    setIsCollapsed(!isCollapsed);
  };

  return (
    <>
      <TouchableOpacity style={styles.ratingsHeading} onPress={onPress}>
        <Text type="small" textType="semiBold" style={styles.textColor}>
          {title}
        </Text>
        <Icon name={isCollapsed ? icons.plus : icons.minus} size={20} color={theme.colors.darkTint4} />
      </TouchableOpacity>
      <Collapsible collapsed={isCollapsed}>{children}</Collapsible>
      <Divider containerStyles={styles.divider} />
    </>
  );
};

class AssetDescription extends React.PureComponent<Props, IOwnState> {
  public state = {
    isFullScreen: false,
    descriptionShowMore: false,
    activeSlide: 0,
    isScroll: true,
  };

  public componentDidMount = (): void => {
    // const { getAssetReviews } = this.props;
    // getAssetReviews(1);
  };

  public render = (): React.ReactNode => {
    const { t, reviews } = this.props;

    return (
      <>
        <StatusBarComponent backgroundColor={theme.colors.white} isTranslucent />
        <ParallaxScrollView
          backgroundColor={theme.colors.white}
          stickyHeaderHeight={STICKY_HEADER_HEIGHT}
          parallaxHeaderHeight={PARALLAX_HEADER_HEIGHT}
          backgroundSpeed={10}
          onChangeHeaderVisibility={(isChanged: boolean): void => this.setState({ isScroll: isChanged })}
          renderForeground={(): React.ReactElement => this.renderCarousel()}
          renderStickyHeader={(): React.ReactElement => this.stickyHeader()}
          renderFixedHeader={(): React.ReactElement => this.fixedHeader()}
        >
          <View style={styles.screen}>
            {this.renderHeaderSection()}
            <CollapsibleSection title={t('description')}>{this.renderAssetDescription()}</CollapsibleSection>
            <CollapsibleSection title={t('highlights')}>{this.renderAssetHighlights()}</CollapsibleSection>
            {this.renderMapSection()}
            <CollapsibleSection title={t('reviewsRatings')}>
              <AssetRatings reviews={reviews} />
            </CollapsibleSection>
          </View>
        </ParallaxScrollView>
        {this.renderFullscreenCarousel()}
      </>
    );
  };

  private renderHeaderSection = (): React.ReactElement | null => {
    const { assetDetails, t } = this.props;
    if (!assetDetails) {
      return null;
    }
    const {
      spaces,
      carpetArea,
      carpetAreaUnit,
      floorNumber,
      assetType,
      leaseTerm,
      saleTerm,
      postedOn,
      availableFrom,
      projectName,
      assetGroup: { name },
    } = assetDetails;
    const propertyType = assetType ? assetDetails.assetType.name : '';
    const term = leaseTerm || saleTerm;

    const amenitiesData: IAmenitiesIcons[] = PropertyUtils.getAmenities(
      carpetArea,
      carpetAreaUnit,
      spaces,
      floorNumber,
      name,
      true
    );

    const propertyTimelineData = PropertyUtils.getPropertyTimelineData(
      name,
      leaseTerm,
      saleTerm,
      postedOn,
      availableFrom
    );

    return (
      <View style={styles.headerContainer}>
        <ShieldGroup text={propertyType} />
        <View style={styles.apartmentContainer}>
          <PricePerUnit price={term.expectedPrice || 0} currency="INR" unit="month" />
          <View style={styles.textIcon}>
            <Icon name={icons.timer} size={22} color={theme.colors.blue} style={styles.iconStyle} />
            <Text type="small" textType="regular" style={styles.primaryText}>
              {t('bookTour')}
            </Text>
          </View>
        </View>
        <View style={styles.apartmentContainer}>
          <PropertyAddress
            isIcon={false}
            primaryAddress="Eaton Garth Manor" // TODO: (Shikha) - remove once api integrate
            subAddress={projectName}
            subAddressStyle={styles.subAddress}
          />
          <View style={styles.textIcon}>
            <View style={styles.verticalDivider} />
            <Icon name={icons.houseMarker} size={30} color={theme.colors.blue} style={styles.iconStyle} />
          </View>
        </View>
        <PropertyAmenities
          data={amenitiesData}
          direction="row"
          containerStyle={styles.amenitiesContainer}
          contentContainerStyle={styles.amenitiesContent}
        />
        <Divider />
        <View style={styles.timelineContainer}>{this.renderPropertyTimelines(propertyTimelineData)}</View>
      </View>
    );
  };

  // TODO: (Shikha) - Add type once api integrate
  private renderPropertyTimelines = (data: any): React.ReactElement => {
    return (
      <>
        {data.map((item: any, index: number) => {
          return (
            <View key={index} style={styles.utilityItem}>
              <Label type="large" textType="regular" style={{ color: theme.colors.darkTint4 }}>
                {item.label}
              </Label>
              <Label type="large" textType="semiBold" style={{ color: theme.colors.darkTint2 }}>
                {item.value}
              </Label>
            </View>
          );
        })}
      </>
    );
  };

  private renderAssetDescription = (): React.ReactNode => {
    const { t, assetDetails } = this.props;
    const { descriptionShowMore } = this.state;

    const onPress = (): void => {
      this.setState({ descriptionShowMore: !descriptionShowMore });
    };

    return (
      <>
        <Label
          type="large"
          textType="regular"
          style={styles.description}
          numberOfLines={descriptionShowMore ? undefined : 3}
        >
          {assetDetails?.description}
        </Label>
        <Label type="large" textType="semiBold" style={styles.helperText} onPress={onPress}>
          {descriptionShowMore ? t('property:showLess') : t('property:showMore')}
        </Label>
      </>
    );
  };

  private renderAssetHighlights = (): React.ReactNode => {
    const { assetDetails } = this.props;
    return (
      <FlatList<AssetHighlight>
        data={assetDetails?.highlights}
        numColumns={2}
        contentContainerStyle={styles.highlightsContainer}
        renderItem={({ item }: { item: AssetHighlight }): React.ReactElement => (
          <View style={styles.highlightItemContainer}>
            <Icon name={icons.check} color={theme.colors.completed} size={22} />
            <Label type="large" textType="regular" style={styles.highlightText}>
              {item.name}
            </Label>
          </View>
        )}
      />
    );
  };

  private renderMapSection = (): React.ReactNode => {
    const { t, assetDetails } = this.props;

    if (!assetDetails) return null;
    const { latitude, longitude } = assetDetails;

    return (
      <View style={styles.neighborhoodContainer}>
        <Text type="small" textType="semiBold" style={styles.textColor}>
          {t('exploreNeighbourhood')}
        </Text>
        <Label type="large" textType="regular" style={styles.neighborhoodAddress}>
          {assetDetails?.projectName}
        </Label>
        <MapView
          pointerEvents="none"
          provider={PROVIDER_GOOGLE}
          style={styles.mapView}
          initialRegion={{
            latitude,
            longitude,
            latitudeDelta: 0.01,
            longitudeDelta: 0.01,
          }}
        >
          <Marker coordinate={{ latitude, longitude }}>
            <CustomMarker selected />
          </Marker>
        </MapView>
        <View style={styles.exploreMapContainer}>
          <Label type="regular" textType="regular" style={styles.exploreMap}>
            {t('exploreMap')}
          </Label>
        </View>
        <Divider containerStyles={styles.divider} />
      </View>
    );
  };

  private renderCarousel = (): React.ReactElement => {
    const { activeSlide } = this.state;
    return (
      <AssetDetailsImageCarousel
        enterFullScreen={this.onFullScreenToggle}
        images={IMAGES}
        activeSlide={activeSlide}
        updateSlide={this.updateSlide}
      />
    );
  };

  private renderFullscreenCarousel = (): React.ReactNode => {
    const { isFullScreen, activeSlide } = this.state;
    if (!isFullScreen) return null;
    return (
      <FullScreenAssetDetailsCarousel
        onFullScreenToggle={this.onFullScreenToggle}
        activeSlide={activeSlide}
        data={IMAGES}
        updateSlide={this.updateSlide}
      />
    );
  };

  private onFullScreenToggle = (): void => {
    const { isFullScreen } = this.state;
    this.setState({ isFullScreen: !isFullScreen });
  };

  private fixedHeader = (): React.ReactElement => {
    const { isScroll } = this.state;
    const color = isScroll ? theme.colors.white : theme.colors.darkTint1;
    return (
      <View key="fixed-header" style={styles.fixedSection}>
        <View style={styles.headerLeftIcon}>
          <Icon name={icons.leftArrow} size={22} color={color} />
        </View>
        <View style={styles.headerRightIcon}>
          <Icon name={icons.heartOutline} size={22} color={color} />
          <Icon name={icons.compare} size={22} color={color} />
        </View>
      </View>
    );
  };

  private stickyHeader = (): React.ReactElement => {
    return (
      <WithShadowView outerViewStyle={styles.shadowView}>
        <View key="sticky-header" style={styles.stickySection}>
          <Text type="regular" textType="semiBold" style={styles.headerTitle}>
            Eaton Garth Manor
          </Text>
        </View>
      </WithShadowView>
    );
  };

  public updateSlide = (slideNumber: number): void => {
    this.setState({ activeSlide: slideNumber });
  };
}

const mapStateToProps = (state: IState): IStateProps => {
  return {
    reviews: AssetSelectors.getAssetReviews(state),
    assetDetails: AssetSelectors.getAsset(state),
  };
};

const mapDispatchToProps = (dispatch: Dispatch): IDispatchProps => {
  const { getAssetReviews } = AssetActions;
  return bindActionCreators({ getAssetReviews }, dispatch);
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withTranslation(LocaleConstants.namespacesKey.assetDescription)(AssetDescription));

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    backgroundColor: theme.colors.white,
    paddingHorizontal: theme.layout.screenPadding,
  },
  headerContainer: {
    borderBottomWidth: 1,
    borderBottomColor: theme.colors.darkTint10,
  },
  ratingsHeading: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: 24,
  },
  highlightsContainer: {
    marginTop: 16,
  },
  highlightItemContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 16,
  },
  textColor: {
    color: theme.colors.darkTint4,
  },
  neighborhoodContainer: {
    marginTop: 24,
  },
  mapView: {
    flex: 1,
    marginTop: 12,
    height: 180,
  },
  divider: {
    marginTop: 24,
  },
  apartmentContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginVertical: 8,
  },
  timelineContainer: {
    paddingVertical: 16,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  iconStyle: {
    marginHorizontal: 6,
  },
  primaryText: {
    color: theme.colors.primaryColor,
  },
  description: {
    color: theme.colors.darkTint4,
    marginTop: 12,
  },
  helperText: {
    marginTop: 12,
    color: theme.colors.active,
  },
  highlightText: {
    color: theme.colors.darkTint4,
    marginStart: 16,
  },
  headerLeftIcon: {
    position: 'absolute',
    left: relativeWidth(2),
  },
  headerRightIcon: {
    flexDirection: 'row',
    position: 'absolute',
    justifyContent: 'space-between',
    right: relativeWidth(4),
    width: 80,
  },
  fixedSection: {
    position: 'absolute',
    width: theme.viewport.width,
    top: 20,
    flexDirection: 'row',
  },
  stickySection: {
    paddingBottom: 10,
    width: 300,
    justifyContent: 'flex-end',
  },
  shadowView: {
    marginTop: 15,
  },
  headerTitle: {
    marginLeft: 40,
    color: theme.colors.darkTint1,
  },
  utilityItem: {
    marginRight: 20,
    marginLeft: 4,
  },
  neighborhoodAddress: {
    marginTop: 12,
    color: theme.colors.darkTint3,
  },
  exploreMapContainer: {
    position: 'absolute',
    bottom: 36,
    alignSelf: 'center',
    paddingHorizontal: 12,
    paddingVertical: 6,
    borderRadius: 4,
    backgroundColor: theme.colors.white,
  },
  exploreMap: {
    color: theme.colors.primaryColor,
  },
  textIcon: {
    flexDirection: 'row',
  },
  subAddress: {
    marginHorizontal: 0,
  },
  verticalDivider: {
    borderWidth: 1,
    height: 30,
    marginRight: 14,
    borderColor: theme.colors.darkTint10,
  },
  amenitiesContainer: {
    justifyContent: 'flex-start',
    marginBottom: 14,
  },
  amenitiesContent: {
    marginRight: 16,
  },
});
