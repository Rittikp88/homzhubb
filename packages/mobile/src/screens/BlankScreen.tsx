import React from 'react';
import { View, StyleSheet, Text } from 'react-native';

export class BlankScreen extends React.PureComponent<{}, {}> {
  public render = (): React.ReactElement => {
    return (
      <View style={styles.screen}>
        <Text>Blank Screen</Text>
      </View>
    );
  };
}

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
