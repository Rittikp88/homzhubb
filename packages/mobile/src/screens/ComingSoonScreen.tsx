import React from 'react';
import { View, StyleSheet } from 'react-native';
import { WithTranslation, withTranslation } from 'react-i18next';
import { DashboardNavigatorParamList, MoreStackNavigatorParamList } from '@homzhub/mobile/src/navigation/BottomTabs';
import { theme } from '@homzhub/common/src/styles/theme';
import { Text } from '@homzhub/common/src/components';
import { AnimatedProfileHeader } from '@homzhub/mobile/src/components';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';

type paramsList = DashboardNavigatorParamList | MoreStackNavigatorParamList;
type libraryProps = NavigationScreenProps<paramsList, ScreensKeys.ComingSoonScreen>;

type Props = WithTranslation & libraryProps;

class ComingSoonScreen extends React.PureComponent<Props> {
  public render = (): React.ReactElement => {
    const {
      t,
      route: {
        params: { title, tabHeader },
      },
    } = this.props;

    return (
      <AnimatedProfileHeader
        title={tabHeader}
        sectionHeader={title}
        onBackPress={this.goBack}
        sectionTitleType="semiBold"
      >
        <View style={styles.screen}>
          <Text type="large" textType="semiBold">
            {t('comingSoonText')}
          </Text>
        </View>
      </AnimatedProfileHeader>
    );
  };

  private goBack = (): void => {
    const { navigation } = this.props;
    navigation.goBack();
  };
}

export default withTranslation(LocaleConstants.namespacesKey.common)(ComingSoonScreen);

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.colors.white,
    paddingTop: theme.viewport.height / 3,
    paddingBottom: theme.viewport.height / 3,
  },
});
