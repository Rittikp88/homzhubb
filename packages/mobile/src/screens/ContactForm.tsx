import React from 'react';
import { View, StyleSheet, SafeAreaView, ScrollView } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { bindActionCreators, Dispatch } from 'redux';
import { connect } from 'react-redux';
import { remove, find, cloneDeep } from 'lodash';
import { AlertHelper } from '@homzhub/mobile/src/utils/AlertHelper';
import { PlatformUtils } from '@homzhub/common/src/utils/PlatformUtils';
import { IState } from '@homzhub/common/src/modules/interfaces';
import { AssetActions } from '@homzhub/common/src/modules/asset/actions';
import { UserSelector } from '@homzhub/common/src/modules/user/selectors';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { ILeadPayload } from '@homzhub/common/src/domain/repositories/interfaces';
import { SearchStackParamList } from '@homzhub/mobile/src/navigation/BottomTabNavigator';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { Label, WithShadowView, Text, Divider, Avatar, Button } from '@homzhub/common/src/components';
import { StatusBarComponent, TimeSlotGroup } from '@homzhub/mobile/src/components';
import { TextArea } from '@homzhub/common/src/components/atoms/TextArea';
import { RadioButtonGroup } from '@homzhub/common/src/components/molecules/RadioButtonGroup';
import { MultipleButtonGroup } from '@homzhub/mobile/src/components/molecules/MultipleButtonGroup';
import { BedroomType, TimeSlot, UserType } from '@homzhub/common/src/mocks/ContactFormData';

interface ISlot {
  from: number;
  to: number;
}

export interface IRadiaButtonGroupData {
  id: number;
  label: string;
}

interface IStateProps {
  isLoggedIn: boolean;
}

interface IDispatchProps {
  postLead: (payload: ILeadPayload) => void;
}

interface IContactState {
  selectedTime: number[];
  selectedSpaces: string[];
  userType: number;
  message: string;
}

type libraryProps = WithTranslation & NavigationScreenProps<SearchStackParamList, ScreensKeys.ContactForm>;
type Props = libraryProps & IDispatchProps & IStateProps;

class ContactForm extends React.PureComponent<Props, IContactState> {
  public state = {
    selectedTime: [],
    selectedSpaces: [],
    userType: 1,
    message: '',
  };

  public render = (): React.ReactElement => {
    const { selectedTime, selectedSpaces, userType } = this.state;
    const { t } = this.props;
    return (
      <>
        <StatusBarComponent backgroundColor="white" isTranslucent={false} statusBarStyle={styles.statusBar} />
        <SafeAreaView style={styles.flexOne}>
          {this.renderHeader()}
          <ScrollView style={styles.scrollView}>
            {this.renderContactDetail()}
            <Divider containerStyles={styles.divider} />
            <Text type="small" textType="semiBold" style={styles.userType}>
              {t('iAm')}
            </Text>
            <RadioButtonGroup
              data={UserType}
              onToggle={this.onUserToggle}
              containerStyle={styles.radioGroup}
              selectedValue={userType}
            />
            <Label type="large" textType="semiBold" style={styles.label}>
              {t('lookingFor')}
            </Label>
            <MultipleButtonGroup
              data={BedroomType}
              onItemSelect={this.onSelectLookingType}
              selectedItem={selectedSpaces}
              containerStyle={styles.buttonGroup}
            />
            <Label type="large" textType="semiBold" style={styles.label}>
              {t('preferredTime')}
            </Label>
            <TimeSlotGroup data={TimeSlot} selectedItem={selectedTime} onItemSelect={this.onTimeSelect} />
            <TextArea
              label={t('message')}
              placeholder={t('typeYourMessage')}
              isOptional
              containerStyle={styles.textArea}
              onMessageChange={this.handleMessageChange}
            />
          </ScrollView>
        </SafeAreaView>
        <WithShadowView outerViewStyle={styles.shadowView}>
          <Button
            type="primary"
            title={t('sendMessage')}
            icon={icons.envelope}
            iconColor={theme.colors.white}
            iconSize={22}
            titleStyle={styles.buttonTitleStyle}
            containerStyle={styles.buttonStyle}
            onPress={this.handleSubmit}
          />
        </WithShadowView>
      </>
    );
  };

  private renderHeader = (): React.ReactElement => {
    const { t } = this.props;
    return (
      <WithShadowView>
        <View style={styles.bottomSheetHeader}>
          <Icon name={icons.close} size={22} color={theme.colors.darkTint3} onPress={this.goBack} />
          <Text type="small" textType="semiBold" style={styles.headerTitle}>
            {t('contact')}
          </Text>
        </View>
      </WithShadowView>
    );
  };

  private renderContactDetail = (): React.ReactElement => {
    const {
      t,
      route: {
        params: { contactDetail },
      },
    } = this.props;

    const number = `${contactDetail.countryCode}${contactDetail.phoneNumber}`;
    return (
      <>
        <Label type="large" textType="semiBold" style={styles.userType}>
          {t('youContacting')}
        </Label>
        <Avatar
          fullName={contactDetail.fullName}
          designation="Owner"
          phoneNumber={number}
          containerStyle={styles.avatarStyle}
        />
      </>
    );
  };

  private onTimeSelect = (id: number): void => {
    const { selectedTime } = this.state;
    const newTime: number[] = cloneDeep(selectedTime);
    let value;
    if (newTime.includes(id)) {
      remove(newTime, (item: number) => item === id);
      value = newTime;
    } else {
      value = newTime.concat(id);
    }

    this.setState({ selectedTime: value });
  };

  private onUserToggle = (id: number): void => {
    this.setState({
      userType: id,
    });
  };

  private onSelectLookingType = (type: string): void => {
    const { selectedSpaces } = this.state;
    const newSpaces: string[] = cloneDeep(selectedSpaces);
    let value;
    if (newSpaces.includes(type)) {
      remove(newSpaces, (item: string) => item === type);
      value = newSpaces;
    } else {
      value = newSpaces.concat(type);
    }
    this.setState({ selectedSpaces: value });
  };

  private handleMessageChange = (value: string): void => {
    this.setState({ message: value });
  };

  private handleSubmit = (): void => {
    const {
      isLoggedIn,
      postLead,
      route: {
        params: { contactDetail, propertyTermId },
      },
    } = this.props;
    if (!contactDetail) return;
    const { userType, selectedTime, selectedSpaces, message } = this.state;
    const personType = UserType.find((item) => item.id === userType);
    const timeSlot: ISlot[] = [];

    selectedTime.forEach((id): any => {
      TimeSlot.forEach((item) => {
        if (item.id === id) {
          const slot = {
            from: item.from,
            to: item.to,
          };
          const slotObject = find(timeSlot, slot);
          if (!slotObject) {
            timeSlot.push(slot);
          }
        }
      });
    });

    if (personType && timeSlot.length > 0) {
      const payload = {
        propertyTermId,
        data: {
          spaces: selectedSpaces,
          contact_person_type: personType.label,
          lead_type: 'MAIL',
          message,
          person_contacted: contactDetail.id,
          preferred_contact_time: timeSlot,
        },
      };
      // TODO: Need to move this logic on Asset Description Screen
      if (isLoggedIn) {
        postLead(payload);
      } else {
        AlertHelper.error({ message: 'Login required' });
      }
    }
  };

  private goBack = (): void => {
    const { navigation } = this.props;
    navigation.goBack();
  };
}

const mapStateToProps = (state: IState): IStateProps => {
  return {
    isLoggedIn: UserSelector.isLoggedIn(state),
  };
};

const mapDispatchToProps = (dispatch: Dispatch): IDispatchProps => {
  const { postLead } = AssetActions;
  return bindActionCreators({ postLead }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(withTranslation()(ContactForm));

const styles = StyleSheet.create({
  flexOne: {
    flex: 1,
  },
  statusBar: {
    height: PlatformUtils.isIOS() ? 50 : 10,
  },
  scrollView: {
    paddingHorizontal: 16,
    backgroundColor: theme.colors.white,
  },
  divider: {
    borderColor: theme.colors.darkTint10,
    marginBottom: 24,
  },
  userType: {
    color: theme.colors.darkTint3,
  },
  radioGroup: {
    marginVertical: 16,
  },
  label: {
    marginVertical: 12,
    color: theme.colors.darkTint3,
  },
  buttonGroup: {
    marginBottom: 18,
  },
  textArea: {
    marginVertical: 20,
  },
  bottomSheetHeader: {
    flexDirection: 'row',
    alignContent: 'center',
    paddingBottom: 20,
    marginHorizontal: 24,
  },
  buttonTitleStyle: {
    marginHorizontal: 12,
  },
  headerTitle: {
    flex: 1,
    textAlign: 'center',
  },
  shadowView: {
    paddingTop: 4,
    marginBottom: PlatformUtils.isIOS() ? 20 : 0,
    paddingBottom: 0,
  },
  buttonStyle: {
    flex: 0,
    flexDirection: 'row-reverse',
    margin: 16,
  },
  avatarStyle: {
    paddingVertical: 18,
  },
});
