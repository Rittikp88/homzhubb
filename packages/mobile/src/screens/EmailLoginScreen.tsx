import React from 'react';
import { StyleSheet, View } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { bindActionCreators, Dispatch } from 'redux';
import { connect } from 'react-redux';
import { IState } from '@homzhub/common/src/modules/interfaces';
import { UserActions } from '@homzhub/common/src/modules/user/actions';
import {
  IEmailLoginPayload,
  ILoginFormData,
  IMobileLoginPayload,
} from '@homzhub/common/src/domain/repositories/interfaces';
import { theme } from '@homzhub/common/src/styles/theme';
import { Header } from '@homzhub/common/src/components/molecules/Header';
import { LoginForm } from '@homzhub/common/src/components/organisms/LoginForm';
import { AuthStackParamList } from '@homzhub/mobile/src/navigation/AuthStack';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';

interface IDispatchProps {
  login: (payload: IEmailLoginPayload | IMobileLoginPayload) => void;
}

type libraryProps = WithTranslation & NavigationScreenProps<AuthStackParamList, ScreensKeys.EmailLogin>;
type Props = IDispatchProps & libraryProps;

class EmailLoginScreen extends React.PureComponent<Props> {
  public render(): React.ReactNode {
    const { t } = this.props;
    return (
      <View style={styles.container}>
        <Header
          icon="left-arrow"
          subTitle={t('auth:logInWithEmail')}
          subTitleType="large"
          subTitleColor={theme.colors.dark}
          onIconPress={this.handleIconPress}
        />
        <LoginForm
          isEmailLogin
          handleForgotPassword={this.handleForgotPassword}
          onLoginSuccess={this.handleLoginSuccess}
        />
      </View>
    );
  }

  private handleIconPress = (): void => {
    const { navigation } = this.props;
    navigation.goBack();
  };

  private handleForgotPassword = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.ForgotPassword);
  };

  private handleLoginSuccess = (values: ILoginFormData): void => {
    const { login } = this.props;

    const emailLoginData: IEmailLoginPayload = {
      action: 'EMAIL_LOGIN',
      payload: {
        email: values.email,
        password: values.password,
      },
    };

    login(emailLoginData);
  };
}

const mapDispatchToProps = (dispatch: Dispatch): IDispatchProps => {
  const { login } = UserActions;
  return bindActionCreators(
    {
      login,
    },
    dispatch
  );
};

export default connect<null, IDispatchProps, WithTranslation, IState>(
  null,
  mapDispatchToProps
)(withTranslation()(EmailLoginScreen));

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: theme.colors.white,
  },
});
