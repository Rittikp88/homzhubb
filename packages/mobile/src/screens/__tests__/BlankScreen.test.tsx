import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BlankScreen } from '@homzhub/mobile/src/screens/BlankScreen';

describe('Blank Screen', () => {
  it('should render the Blank screen', () => {
    const component = shallow(<BlankScreen />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
