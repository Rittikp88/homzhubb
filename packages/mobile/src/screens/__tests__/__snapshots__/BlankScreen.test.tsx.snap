// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blank Screen should render the Blank screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Blank Screen
  </Text>
</View>
`;
